tasks/set_vars.yml

```
@@ -5,3 +5,7 @@
     gather_subset: min
   when: not ansible_facts.keys() | list |
     intersect(__rhc_required_facts) == __rhc_required_facts
+
+- name: Check if insights-client is installed
+  package_facts:
+  when: rhc_insights.state | d("present") == "absent"
```

this needs to match the condition of the `Handle insights unregistration` task:
```suggestion
  when:
    - ansible_distribution == "RedHat"
    - >-
      rhc_insights.state | d("present") == "absent"
      or rhc_state | d("present") == "absent"
```

---

Oh right! The tests have change and didn't change the `rhc_insights.state` but the `rhc_state`.